pandas/io/packers.py

```diff
@@ -595,11 +595,14 @@ def decode(obj):
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
 
+        # see gh-14901
+        klass_dtype = np_dtype if pd_dtype.name != 'category' else pd_dtype
+
         index = obj[u'index']
         result = globals()[obj[u'klass']](unconvert(obj[u'data'], dtype,
                                                     obj[u'compress']),
                                           index=index,
-                                          dtype=np_dtype,
+                                          dtype=klass_dtype,
```

Comment:
Here is the correct way to re-serialize.

```
In [29]: Series(s.values, dtype='M8[ns]').dt.tz_localize('UTC').dt.tz_convert(s.dt.tz)
Out[29]:
0   2013-01-02 00:00:00-05:00
1   2013-01-02 00:00:00-05:00
2   2013-01-02 00:00:00-05:00
3   2013-01-02 00:00:00-05:00
4   2013-01-02 00:00:00-05:00
dtype: datetime64[ns, US/Eastern]
````

The issue is you are storing UTC timezones. But you need to then re-convert them back to the correct timezone. This is a standard idiom (e.g. HDFStore does this).